src/walks.jl

```diff
@@ -234,7 +234,17 @@ julia> collect(zipped_iter)
 struct IterateWalk <: AbstractWalk end
 
 function (walk::IterateWalk)(recurse, x, ys...)
-  func, _ = functor(x)
-  yfuncs = map(y -> functor(typeof(x), y)[1], ys)
-  return Iterators.flatten(_map(recurse, func, yfuncs...))
+  x_children = children(x)
+  ys_children = map(children, ys)
+  return Iterators.flatten(_map(recurse, x_children, ys_children...))
 end
+
+struct FlattenWalk <: AbstractWalk end
+
+function (walk::FlattenWalk)(recurse, x, ys...)
+  x_children = _values(children(x))
+  ys_children = map(children, ys)
+  res = _map(recurse, x_children, ys_children...)
+  return reduce(vcat, _values(res))
```

Comment:
Note that this works pairwise, it does not use any of the magic fast methods of reduce, unless you have a vector of objects:
```julia
julia> @eval Functors function (walk::FlattenWalk)(recurse, x, ys...)
         x_children = _values(children(x))
         ys_children = map(children, ys)
         res = _map(recurse, x_children, ys_children...)
         @show typeof(_values(res))
         return reduce(vcat, _values(res))
       end

julia> fleaves((x=(1,2,3), y=(a=4, b=5), z=6))
typeof(_values(res)) = Tuple{Vector{Int64}, Vector{Int64}, Vector{Int64}}
typeof(_values(res)) = @NamedTuple{a::Vector{Int64}, b::Vector{Int64}}
typeof(_values(res)) = @NamedTuple{x::Vector{Int64}, y::Vector{Int64}, z::Vector{Int64}}
6-element Vector{Int64}:
 1
 2
 3
 4
 5
 6

julia> @which reduce(vcat, ([1], [2], [3]))
reduce(op, itr; kw...)
     @ Base reduce.jl:490
```